codes/config.py

```
@@ -38,3 +38,32 @@ def __getitem__(self, key: str):
+                self.trust_penalty_weight /= trust_sum
+        # Q-learning checks
+        if not (0.0 < self.learning_rate <= 1.0):
+            raise ValueError(f"learning_rate must be in (0.0, 1.0], got {self.learning_rate}")
+        if not (0.0 < self.gamma <= 1.0):
+            raise ValueError(f"gamma (discount factor) must be in (0.0, 1.0], got {self.gamma}")
+        if not (0.0 <= self.epsilon_min <= 1.0):
+            raise ValueError(f"epsilon_min must be in [0.0, 1.0], got {self.epsilon_min}")
+        if not (0.0 <= self.epsilon_start <= 1.0):
+            raise ValueError(f"epsilon_start must be in [0.0, 1.0], got {self.epsilon_start}")
+        if not (0.0 < self.epsilon_decay <= 1.0):
+            raise ValueError(f"epsilon_decay must be in (0.0, 1.0], got {self.epsilon_decay}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)